core/store/src/flat_state.rs

```diff
@@ -81,7 +100,7 @@ mod imp {
         /// could charge users for the value length before loading the value.
         // TODO (#7327): support different roots (or block hashes).
         // TODO (#7327): consider inlining small values, so we could use only one db access.
-        pub fn get_ref(&self, key: &[u8]) -> Result<Option<ValueRef>, StorageError> {
+        pub fn get_ref(&self, key: &[u8]) -> Result<Option<ValueRef>, crate::StorageError> {
             // Take deltas ordered from `self.block_hash` to flat state head.
             // In other words, order of deltas is the opposite of the order of blocks in chain.
             let deltas = self.flat_storage_state.get_deltas_between_blocks(&self.block_hash)?;
```

Comment:
Will we move this to the (maybe lazy) initialization of FlatState? I don't imagine computing this for every single key would be very efficient.

Reply:
Yes, but flat_head could change during the lifetime of FlatState, so we can't remove the need for calling get_deltas_between_blocks all together. One optimization we could do is to store the value of flat_head and the deltas path last time when get_deltas_between_blocks is called and only recompute it if flat_head is changed. I'll create a JIRA issue to track this.

Reply:
I see, thanks. As a general feedback here I find the names a bit overloaded. Maybe some documentation text on how each of these structures are different / related could be helpful. For example, it's almost impossible to tell what the difference is between FlatStorageState and FlatState by just looking at the names (it's still fuzzy to me even after reading the code).